### PR TITLE
Update to TS 5.6

### PR DESCRIPTION
**Pull request by mikearnaldi**
Had to update `@types/node` and add `node` as type entry in the bun package to make `Buffer` a valid `UInt8Array` due to the Iterator methods that have been added in 5.6, also had to force some `!` on iterators, else all seems to work.

Notable improvements in LSP performance on files such as "core.ts"

### CODE REVIEW

**Comment by mikearnaldi** [comment]
That's odd it actually doesn't work but apparently building twice makes the error go away

**Review by tim-smart — APPROVED** [review]
Everything seems to look ok 👍🏼

File: packages/platform-bun/tsconfig.src.json [hunk 1]
```diff
@@ -6,6 +6,6 @@
     "tsBuildInfoFile": ".tsbuildinfo/src.tsbuildinfo",
     "rootDir": "src",
     "outDir": "build/src",
-    "types": ["bun-types"]
+    "types": ["bun-types", "node"]
```
Comment (fubhy) [inline packages/platform-bun/tsconfig.src.json:9]: We should remove this again when possible.

**Review by fubhy — APPROVED** [review]

File: packages/platform-bun/tsconfig.src.json (hunk 1 above)
Comment (mikearnaldi) [reply packages/platform-bun/tsconfig.src.json]: Done, it was not needed anyway

**Pull request merged**